tracee/tracee.go

```diff
@@ -997,7 +999,8 @@ type context struct {
 	EventID  int32
 	Retval   int64
 	Argnum   uint8
-	_        [7]byte //padding
+	StackID  uint32
+	_        [3]byte //padding
```

Comment:
Not sure this padding is now correct.
It was 7 bytes when we had one byte at the end,
and the struct needs to be 8 bytes aligned.
Now I think the padding should be 4.
We had some issues with this in the passed which made @itaysk set this padding

Reply:
Is that correct? So on the C side, `context_t` with the new `stack_id` looks like this:
```
typedef struct context {
    u64 ts;                     // Timestamp
    u32 pid;                    // PID as in the userspace term
    u32 tid;                    // TID as in the userspace term
    u32 ppid;                   // Parent PID as in the userspace term
    u32 host_pid;               // PID in host pid namespace
    u32 host_tid;               // TID in host pid namespace
    u32 host_ppid;              // Parent PID in host pid namespace
    u32 uid;
    u32 mnt_id;
    u32 pid_id;
    char comm[16];
    char uts_name[16];
    u32 eventid;
    s64 retval;
    u8 argnum;
    u32 stack_id;
} context_t;
```
So thats `(8 + 4 + 4 + 4 + 4 + 4 + 4 + 4 + 4 + 4 + 16 + 16 + 4 + 8 + 1 + 4) == 93 bytes`. And I checked with a debug string that `sizeof(context_t) == 96`, i.e. has a 3 Byte padding

Reply:
you're logic makes sense. when working on this I remember learning that the napkin calculation sometimes failed me, so I eventually added a debug statement, saw the memory layout and reverse-explained how the padding worked. I'm sorry I don't remember what was the particular caveat that made my original calc fail, but If you confirmed it in practice, it should be good. my method was to add a print in the go code (today that would be here https://github.com/aquasecurity/tracee/blob/main/tracee/pipeline.go#L50), and print a hex-dump of the event, yours sounds good as well.

Reply:
@yanivagman do we need any special casing for ARM? I don't know how it works there just asking

Reply:
From what I remember when I wrote the above comment, StackId (uint32) was the last struct field before the padding, and not the Argnum(u8).
Padding, as far as I remember, work as follows:
1. Each struct should be aligned to its biggest field size (in our case, u64 which is 8 bytes)
2. Each field of the struct should be aligned according to it's size (thus, u32 is 4 bytes aligned)

So, when StackId was the last field before the padding, it was 4 bytes aligned, thus needed 4 bytes padding to align the struct (to 8 bytes). After Argnum is moved to be before the padding, there is a needed to add 3 more bytes to align to 8 bytes, as required by the struct - so it should also be fine